src/main/java/com/iota/iri/TransactionValidator.java

```diff
@@ -102,12 +178,28 @@ public void runValidation(TransactionViewModel transactionViewModel, final int m
         }
     }
 
+    /**
+     * Creates a new transaction from  {@code trits} and validates it with {@link #runValidation}
+     *
+     * @param trits raw transaction trits
+     * @param minWeightMagnitude minimal number of trailing 9s in transaction for POW validation
+     * @return the transaction resulting from the raw trits
```

Comment:
only if valid, otherwise it will throw.

Reply:
The next line `@throws` says this. I added "only if valid"